### PR TITLE
Bonus fe UI enhancement

### LINKED ISSUE #13: Bonus FE: UI Enhancement
Look for areas of visual enhancement in the UI. Animations, loading screens, etc.

### PR DESCRIPTION
**Pull request by samsoncsyu7777**
### What this PR does (required):
- add Grow animation on EditMenu
- add Slide transition on Requests page
- add Collapse animation on Login and Signup pages
- add text animation in LoggedInBar
- add color transition in Edit-Photo component

### Screenshots / Videos (front-end only):

https://user-images.githubusercontent.com/30422337/143677781-5ecab147-4807-440b-9b2c-ab7cebbfe8b8.mp4

https://user-images.githubusercontent.com/30422337/143677783-4fae0260-b7a7-4893-8912-e505b8e676e0.mp4

https://user-images.githubusercontent.com/30422337/143677785-2c7102dd-38fe-41f8-869e-7cb1176ecb10.mp4

https://user-images.githubusercontent.com/30422337/143677786-27c91488-1155-4b01-94ed-d68ffae8f5aa.mp4

https://user-images.githubusercontent.com/30422337/143677787-fa0ea780-ae1e-4c41-b61d-60ba6351b511.mp4

https://user-images.githubusercontent.com/30422337/143677788-95374dfc-202a-4982-ad16-a19bb3c73218.mp4

### Any information needed to test this feature (required):
- npm start

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
Nice bonus feature :+1:

File: client/src/components/NavBar/AuthBars/LoggedInBar.tsx [hunk 1]
```diff
@@ -5,28 +5,63 @@ import { Link } from 'react-router-dom';
...
+    }
+  }, [showText2]);
 
   return (
     <Grid container className={classes.navButtons}>
       <Grid item>
-        <NotificationLink />
+        <NotificationLink>
+          <Typography variant="h3">
+            <TextTransition text={showText2 ? 'Notifications' : ''} springConfig={presets.default} />
```
Comment (rajivtitus) [inline client/src/components/NavBar/AuthBars/LoggedInBar.tsx:44]: Optional: These transition effects look cool but I would suggest reserving them only for the logo (also, the may need to be disabled on mobile views) 😄

File: client/src/context/useFindSitterFormContext.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { FindSitterForm } from '../interface/FindSitterForm';
+import { createContext, FunctionComponent, useState, useContext, useCallback } from 'react';
+
+interface IFindSitterFormContext {
```
Comment (rajivtitus) [inline client/src/context/useFindSitterFormContext.tsx]:
```suggestion
interface FindSitterFormContext {
```
You don't need to prefix interfaces with 'I' in JS :)

File: client/src/components/EditProfile/EditPhoto/useStyles.ts [hunk 3]
```diff
@@ -16,6 +17,14 @@ const useStyles = makeStyles((theme) => ({
   reminder: {
     padding: theme.spacing(2.5),
     width: '16vw',
+    color: '#ff0000',
+    animation: '$colorAnimate 10s infinite',
+  },
+  '@keyframes colorAnimate': {
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditPhoto/useStyles.ts:23]: You can stick to using the primary/secondary colours of the app :)

**Review by rajivtitus — APPROVED** [review]
Excellent work making using of the MUI animation components 😄

File: client/src/components/NavBar/AuthBars/LoggedInBar.tsx (hunk 1 above)
Comment (samsoncsyu7777) [reply client/src/components/NavBar/AuthBars/LoggedInBar.tsx:44]: OK, I disable it on mobiles

File: client/src/context/useFindSitterFormContext.tsx (hunk 2 above)
Comment (samsoncsyu7777) [reply client/src/context/useFindSitterFormContext.tsx]: Thanks for this information. Done.

File: client/src/components/EditProfile/EditPhoto/useStyles.ts (hunk 3 above)
Comment (samsoncsyu7777) [reply client/src/components/EditProfile/EditPhoto/useStyles.ts:23]: Done.

**Pull request merged**